modules/objdetect/src/qrcode.cpp

```diff
@@ -3993,13 +4036,504 @@ bool QRCodeDetector::detectAndDecodeMulti(
+            if (max(abs(dist.x), abs(dist.y)) < 3.f * tmp.moduleSize) {
+                isUniq = false;
+                break;
+            }
+        }
+        if (isUniq) {
+            patterns.push_back(pattern);
+            maxModuleSize = max(maxModuleSize, patterns.back().moduleSize);
+        }
+    }
+    const int threshold = cvRound(qrDetectorParameters.minModuleSizeInPyramid * 12.5f) +
+                          (cvRound(qrDetectorParameters.minModuleSizeInPyramid * 12.5f) % 2 ? 0 : 1);
+    int maxLevelPyramid = 0;
+    while (maxModuleSize / 2.f > qrDetectorParameters.minModuleSizeInPyramid) {
+        maxLevelPyramid++;
+        maxModuleSize /= 2.f;
+    }
+    vector<Mat> pyramid;
+    buildPyramid(img, pyramid, maxLevelPyramid);
+    // TODO: ADAPTIVE_THRESH_GAUSSIAN_C vs ADAPTIVE_THRESH_MEAN_C
+    for (Mat& pyr: pyramid) {
+        adaptiveThreshold(pyr, pyr, 255, ADAPTIVE_THRESH_GAUSSIAN_C, THRESH_BINARY, threshold, -1);
+    }
+
+    for (size_t i = 0ull; i < patterns.size(); i++) {
+        QRCode qrCode;
+        int indexes[3] = {0};
+        for (size_t j = i + 1ull; j < patterns.size(); j++) {
+            for (size_t k = j + 1ull; k < patterns.size(); k++) {
+                Point3i index((int)i, (int)j, (int)k);
+                QRCode tmp = QRCode::checkCompatibilityPattern(patterns[i], patterns[j], patterns[k], index, qrDetectorParameters);
+                if (tmp.distance != std::numeric_limits<float>::max()) {
+                    int levelPyramid = 0;
+                    QRCode qrCopy = tmp;
+                    while (tmp.moduleSize / 2.f > qrDetectorParameters.minModuleSizeInPyramid) {
+                        tmp *= 0.5f;
+                        levelPyramid++;
+                    }
+                    qrCopy.timingPatternScore = tmp.calculateScoreByTimingPattern(pyramid[levelPyramid], qrDetectorParameters);
+                    if (qrCopy.timingPatternScore != std::numeric_limits<int>::max() &&
+                        qrCopy.timingPatternScore * qrDetectorParameters.scaleTimingPatternScore < (float)qrCode.timingPatternScore
+                        && qrCopy.distance < qrCode.distance)
+                    {
+                        qrCode = qrCopy;
+                        indexes[0] = (int)i;
+                        indexes[1] = (int)j;
+                        indexes[2] = (int)k;
+                    }
+                }
+            }
+        }
+        if (qrCode.distance != std::numeric_limits<float>::max()) {
+            qrCodes.push_back(qrCode);
+            std::swap(patterns[indexes[2]], patterns.back());
+            patterns.pop_back();
+            std::swap(patterns[indexes[1]], patterns.back());
+            patterns.pop_back();
+            std::swap(patterns[indexes[0]], patterns.back());
+            patterns.pop_back();
+            i--;
+        }
+    }
+    return qrCodes;
+}
+
+struct PimplQRAruco : public ImplContour {
```

Comment:
`detect` and `detectAndDecode` are not overloaded. It means, that base class method is used. It's large confusion for users, because provided parameters are not applied. I propose just to call `detectMulti` and select largest QR or QR with highest confidence.